Libraries/LibC/string.cpp

```diff
@@ -457,9 +457,50 @@ char* strpbrk(const char* s, const char* accept)
 
 char* strtok(char* str, const char* delim)
 {
-    (void)str;
-    (void)delim;
-    ASSERT_NOT_REACHED();
+    static char* saved_str;
+    
+    if (!str) {
+        if (!saved_str)
+            return nullptr;
+        str = saved_str;
+    }
+    
+    size_t token_start = 0;
+    size_t token_end = 0;
+    
+    for (size_t i = 0; i < strlen(str); ++i) {
+        bool is_delim = false;
+        for (size_t j = 0; j < strlen(delim); ++j) {
+            if (str[i] == delim[j]) {
+                is_delim = true;
+                ++token_start;
+                break;
+            }
+        }
+        if (!is_delim) break;
+    }
+    
+    if (strlen(&str[token_start]) == 0)
+        return nullptr;
+    
+    for (size_t i = 0; i < strlen(delim); ++i) {
+        char* delim_ptr = strchr(&str[token_start], delim[i]);
```

Comment:
This seems really aggressive on the string scanning, let's find a way to do it without `strchr()` :)

Reply:
I don't think I understand what's aggressive about strchr. I would have to scan the string until a delimiter anyways, and strchr takes care of that pretty efficiently. Would you mind elaborating so I better understand the issue with strchr? Thanks!

Reply:
Do you want something like this within the for loop? This approach works in place of strchr

```cpp
bool delim_found = false;
size_t str_index = token_start;
while (str[str_index] != '\0') {
    if (str[str_index] == delim[i]) {
        delim_found = true;
        break;
    }
    ++str_index;
 }

if ((delim_found) && (!token_end || token_end > str_index))
    token_end = str_index;
```

Reply:
Sorry, I should have been more specific!
I meant it's quite aggressive in the situation where you have multiple delimiters and a long potential token string to scan through.
For each not-found token, you end up scanning from token_start to the end of the string.
If we assume that some delimiter is going to be found, we can walk through the string in the outer loop, and check against all delimiter chars in the inner loop.
Does that make sense? Or maybe my intuition about this is off and I'm missing something? :)

Reply:
Okay, I see what you're saying now! When I have some free time I'll go in and get that figured out. Thank you! :)